src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -306,6 +314,196 @@ void DatasetManager::HandleGet(Coap::Header &aHeader, Message &aMessage, const I
+    }
+
+    if (aDataset.mIsDelaySet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_DELAYTIMER);
+        tlv.SetLength(sizeof(aDataset.mDelay));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data32 = HostSwap32(aDataset.mDelay);
+        SuccessOrExit(error = message->Append(&data32, sizeof(aDataset.mDelay)));
+    }
+
+    if (aDataset.mIsPanIdSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_PANID);
+        tlv.SetLength(sizeof(aDataset.mPanId));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data16 = HostSwap16(static_cast<uint16_t>(aDataset.mPanId));
+        SuccessOrExit(error = message->Append(&data16, sizeof(aDataset.mPanId)));
+    }
+
+    if (aDataset.mIsChannelSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_CHANNEL);
+        tlv.SetLength(sizeof(aDataset.mChannel));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data16 = HostSwap16(aDataset.mChannel);
+        SuccessOrExit(error = message->Append(&data16, sizeof(aDataset.mChannel)));
+    }
+
+    if (aSize > 0)
+    {
+        SuccessOrExit(error = message->Append(aTlvs, aSize));
+    }
+
+    mMle.GetLeaderAddress(leader);
+
+    memset(&messageInfo, 0, sizeof(messageInfo));
+    memcpy(&messageInfo.mPeerAddr, &leader, sizeof(messageInfo.mPeerAddr));
+    messageInfo.mPeerPort = kCoapUdpPort;
+    SuccessOrExit(error = mSocket.SendTo(*message, messageInfo));
+
+    otLogInfoMeshCoP("sent dataset set request to leader\n");
+
+exit:
+
+    if (error != kThreadError_None && message != NULL)
+    {
+        Message::Free(*message);
+    }
+
+    return error;
+}
+
+ThreadError DatasetManager::SendGetRequest(const uint8_t *aTlvTypes, const uint8_t aSize)
+{
+    ThreadError error = kThreadError_None;
+    Coap::Header header;
+    Message *message;
+    Ip6::Address leader;
+    Ip6::MessageInfo messageInfo;
+    Tlv tlv;
+
+    mSocket.Open(&HandleUdpReceive, this);
+
+    for (size_t i = 0; i < sizeof(mCoapToken); i++)
+    {
+        mCoapToken[i] = static_cast<uint8_t>(otPlatRandomGet());
+    }
+
+    header.Init();
+    header.SetVersion(1);
+    header.SetType(Coap::Header::kTypeConfirmable);
+    header.SetCode(Coap::Header::kCodePost);
+    header.SetMessageId(++mCoapMessageId);
+    header.SetToken(mCoapToken, sizeof(mCoapToken));
+    header.AppendUriPathOptions(mUriGet);
+    header.AppendContentFormatOption(Coap::Header::kApplicationOctetStream);
+    header.Finalize();
+
+    VerifyOrExit((message = Ip6::Udp::NewMessage(0)) != NULL, error = kThreadError_NoBufs);
+    SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
+
+    if (aSize > 0)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_GET);
+        tlv.SetLength(aSize);
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        SuccessOrExit(error = message->Append(aTlvTypes, aSize));
+    }
+
+    mMle.GetLeaderAddress(leader);
```

Comment:
You can write the address directly into `messageInfo` and remove the need for having `leader`.